server/utils/paginationUtils.ts

```diff
@@ -0,0 +1,123 @@
+/*
```

Comment:
I found a pure Javascript version of a pagination service and transcribed to typescript and tidied a little.